### ISSUE TITLE
error on winbuilder machine re magnetiField()

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1718]
`devtools::check_win()` on the "develop" branch yields error
```R
  Linking to GEOS 3.8.0, GDAL 3.0.4, PROJ 6.3.1
  > 
  > test_check("oce")
  -- 1. Failure: magneticField() handles both POSIX times and dates (@test_misc.R#
  `A` not equal to `B`.
  1/1 mismatches
  [1] -18 - -18 == -8.93e-05
  
  == testthat results  ===========================================================
  [ OK: 2560 | SKIPPED: 41 | WARNINGS: 0 | FAILED: 1 ]
  1. Failure: magneticField() handles both POSIX times and dates (@test_misc.R#259) 
```
but when I run the code
```R
>  A <- magneticField(-63.562, 44.640, as.POSIXct("2013-01-01"), version=12)$declination
>  B <- magneticField(-63.562, 44.640, as.Date("2013-01-01"), version=12)$declination
> identical(A, B)
[1] TRUE
```
locally, I get identical answers for `A` and `B`.  Perhaps this is a mac-vs-windows thing (but, if so, why no error in past years?) or perhaps it's a locale thing (but, again, why no error in past years?)

**Comment by richardsc** [comment]
I can check this on my machine, but first I need to make sure I have all the needed libraries up-to-date

**Comment by dankelley** [comment]
Hm. `magneticField()` In `misc.R` does
```R
    if (inherits(time, "Date"))
        time <- as.POSIXct(time, tz="UTC")
```
but that (for me, in Halifax today) gives a 4-hour mismatch.  Maybe the windbuilder machine is in a timezone that's different enough (more than 4 hours) to get past the threshold of the test on magnetic field.

In any case, I think the conversion should be as if the `Date` item was in UTC, *whether the user did that or not*.

**Comment by dankelley** [comment]
Hm.  I get as below
```R
> t1 <- as.Date("2013-01-01")
> format(t0,"%Y-%m-%d %H:%M:%S %z")
[1] "2013-01-01 00:00:00 -0400"
> dput(t1)
structure(15706, class = "Date")
```
which confuses me a bit.  It seems that the timezone is not burned into the `Date` object (from the `dput` output) but that maybe the tz is coming from the locale later.

Maybe I can solve this by using `lubridate` but I am a bit reluctant to add a dependence on another package just for this, when the solution is going to end up being a one-liner (I think).

**Comment by dankelley** [comment]
EDIT: my last comment was wrong.  I was looking at the `%z` of `t0`, not `t1`.

I now think that the answer is that my test is simply wrong.  I ought to be doing
```R
A <- magneticField(-63.562, 44.640, as.POSIXct("2013-01-01", tz="UTC"), version=12)$declination
B <- magneticField(-63.562, 44.640, as.Date("2013-01-01"), version=12)$declination
```

**Comment by dankelley** [comment]
Oh, yes, doing that `tz="UTC"` works.  See https://win-builder.r-project.org/Iqbd0s9ER653/00check.log which is a test page that will disappear in a day or so, I think.  Ignore the notes that are given there ... these will be fixed up when I merge "sf" into "develop" (since I accidentally fixed those links in the "sf" branch instead of in "develop").

**Comment by dankelley** [comment]
Fixed in "develop" commit bcb6b5e97bf8f8418ba32fd698a914f497349e55
